src/nvim/main.c

```diff
@@ -921,6 +921,10 @@ static void remote_request(mparm_T *params, int remote_args, char *server_addr,
       os_errmsg(connect_error);
       os_errmsg("\n");
       os_exit(1);
+    } else if (strequal(server_addr, os_getenv("NVIM"))) {
+      os_errmsg("Cannot attach UI of :terminal child to its parent. (Unset $NVIM to skip this check)");
```

Comment:
Lint failure because this message is too long.
Also should this message be translated? @justinmk